platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/ConnectionServer.java

```diff
@@ -102,4 +109,26 @@ private void listen(final ServerSocket serverSocket) throws InterruptedException
             }
         }
     }
+
+    /**
+     * Adds a handshake completed listener
+     * */
+    private void addHandshakeListener(final Socket clientSocket) {
```

Comment:
this should be used for identification, but right now it just closes the socket if it cant match a peer

Reply:
This is the expected behavior by my understanding. Do we want to leave the client socket open when we can't match it to a peer?

Reply:
we want to close the socket, that part is fine. the issue is that it should be used for identifying the peer instead of the current method or reading the public key from the socket